File: src/dataset_creation.py
import os
import pandas as pd

def create_dataset(path="data/bbc"):
    categories = [ name for name in os.listdir(path) if os.path.isdir(os.path.join(path, name)) ]
    
    data = []
    for cat in categories:
        cat_path = os.path.join(path, cat)
        files = os.listdir(cat_path)
        for file in files:
            file_path = os.path.join(cat_path, file)
            f = open(file_path, 'r', encoding="iso-8859-1")
            data.append([file, ''.join(f.readlines()), cat])
    df = pd.DataFrame(columns=['File_Name', 'Content', 'Category'], data=data)
    df.to_csv("data/datasets/contents.csv")

if __name__ == '__main__':
    create_dataset()

File: src/dataset_creation.r
#################################################################################################
# 01. DATASET CREATION

# The aim of this script is to create a dataset with the following information:
#   - Name of the article
#   - Content of the article
#   - Category of the article
#################################################################################################
# Installs
#install.packages("readtext", dependencies=T)

# Imports
library(readtext)

# Cleaning environment data
rm(list = ls())

# Working directory
#setwd('')

# Path definition of the news archives
path <- 'data/bbc-fulltext/bbc'

# List with the 5 categories
list_categories <- list.files(path=path)

print(list_categories)

# à terminer. c'était pour m'assurer que le csv produit par dataset_creation.py était le même que celui-ci (de l'auteur de l'article) mais pas le temps.